extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/PgPoolCreator.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.reactive.pg.client;
+
+import io.vertx.core.Vertx;
+import io.vertx.pgclient.PgConnectOptions;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.PoolOptions;
+
+/**
+ * This interface is an integration point that allows users to use {@link Vertx}, {@link PoolOptions} and
+ * {@link PgConnectOptions}
+ * configured automatically by Quarkus, but provide on a custom strategy for creating the final {@link PgPool}, instead of
+ * letting Quarkus create it (by calling {@link PgPool#pool(Vertx, PgConnectOptions, PoolOptions)}).
+ *
+ * Implementations of this class are meant to be used as CDI beans - if a single implementation marked as a CDI bean,
+ * it will be picked up by Quarkus and used to create the pool.
+ */
+public interface PgPoolCreator {
+
+    PgPool create(Input input);
+
+    interface Input {
+
+        Vertx vertx();
+
+        PoolOptions poolOptions();
+
+        PgConnectOptions pgConnectOptions();
+    }
+}
```

Comment:
```suggestion
public interface PgPoolCreator {
    PgPool create(PgPoolOptions poolOptions, PgConnectOptions connectOptions);
}
```

Reply:
I think this can be simplified to a method which takes two params (which must be the option types specific to PostgreSQL, not the base ones).

This is assuming the Vert.x instance can be `@Inject`-ed in the user-provided creator.

Reply:
Yeah, we can certainly drop the Vertex part, but I want to keep the extra Input class in case we find in the future that adding more context makes sense.

Reply:
Actually, I don't think this would be a good user experience as users need to figure out where to get the required Vertx instance as opposed to just looking at the method input

Reply:
OK. I guess it's fine to have an additional object given this call is definitely not critical.

Reply:
Right, it only gets created once (if at all)